lib/src/wolt_modal_sheet.dart

```diff
@@ -424,4 +533,231 @@ class _WoltModalSheetState extends State<WoltModalSheet> {
+  void addPagesToStack(List<SliverWoltModalSheetPage> pages) {
+    setState(() {
+      _pages.addAll(pages);
+    });
+  }
+
+  /// Overload of addPagesToStack for adding a single page.
+  ///
+  /// Parameters:
+  /// - [page]: The single [SliverWoltModalSheetPage] to be added to the stack.
+  void addPageToStack(SliverWoltModalSheetPage page) {
+    addPagesToStack([page]);
+  }
+
+  /// Dynamically updates the navigation stack by adding new pages or replacing subsequent pages
+  /// depending on the current page's position in the stack.
+  ///
+  /// If the current page is the last in the stack, this method appends the new pages. If the current
+  /// page is not the last, it replaces all pages after the current one with the provided list of new pages.
+  /// This approach is particularly useful in scenarios where the user's navigation path needs dynamic
+  /// updates based on interactions on the last page or when revising choices made in previous steps.
+  ///
+  /// Parameters:
+  /// - [pages]: The list of [SliverWoltModalSheetPage] to be appended or to replace existing pages.
+  ///   This can be a single page or multiple pages.
+  ///
+  /// Returns:
+  /// - [bool]: Returns `true` if the operation affects the stack, `false` if the stack is empty
+  ///   and no operation was performed.
+  bool addOrReplacePages(List<SliverWoltModalSheetPage> pages) {
+    setState(() {
+      if (_selectedPageIndex == _pages.length - 1) {
+        // Append new pages if the current page is the last one.
+        _pages.addAll(pages);
+      } else {
+        // Replace all pages beyond the current one with new pages.
+        _pages = List<SliverWoltModalSheetPage>.from(_pages.take(_selectedPageIndex + 1))
+          ..addAll(pages);
+      }
+    });
+
+    return true; // The operation always affects the stack if it's not empty.
+  }
+
+  /// Overload of addOrReplaceLastPages for adding a single page.
+  ///
+  /// Parameters:
+  /// - [page]: The single [SliverWoltModalSheetPage] to be added or to replace the last page.
+  bool addOrReplacePage(SliverWoltModalSheetPage page) {
+    return addOrReplacePages([page]);
+  }
+
+  /// Adds one or more new pages to the modal sheet stack and displays the first of the added
+  /// page list.
+  ///
+  /// This method pushes a list of [SliverWoltModalSheetPage] onto the navigation stack,
+  /// adding them to the end. It then updates the displayed page to be the first page in the newly added list,
+  /// allowing for a sequence of pages to be added and immediately interacted with.
+  ///
+  /// Parameters:
+  /// - [pages]: The list of [SliverWoltModalSheetPage] to add to the navigation stack.
+  ///
+  /// Returns:
+  /// None.
+  void pushPages(List<SliverWoltModalSheetPage> pages) {
+    setState(() {
+      _pages.addAll(pages);
+      if (pages.isNotEmpty) {
+        _selectedPageIndex =
+            _pages.length - pages.length; // Set to the first of the newly added pages.
+      }
```

Comment:
```suggestion
        _selectedPageIndex =
            _pages.length - pages.length; // Set to the first of the newly added pages.
```

if we're checking `pages` isn't empty above, we can feel confident `page` is not empty and update this.